zuul-core/src/main/java/com/netflix/zuul/netty/server/DirectMemoryMonitor.java

```diff
@@ -94,35 +51,33 @@ public final class DirectMemoryMonitor {
 
     @Inject
     public DirectMemoryMonitor(Registry registry) {
-        if (reservedMemoryGetter != null) {
-            PolledMeter.using(registry)
-                    .withName(PROP_PREFIX + ".reserved")
-                    .withDelay(Duration.ofSeconds(TASK_DELAY_PROP.get()))
-                    .scheduleOn(service)
-                    .monitorValue(DirectMemoryMonitor.class, DirectMemoryMonitor::getReservedMemory);
-        }
-        if (directMemoryLimitGetter != null) {
-            PolledMeter.using(registry)
-                    .withName(PROP_PREFIX + ".max")
-                    .withDelay(Duration.ofSeconds(TASK_DELAY_PROP.get()))
-                    .scheduleOn(service)
-                    .monitorValue(DirectMemoryMonitor.class, DirectMemoryMonitor::getMaxMemory);
-        }
+        PolledMeter.using(registry)
+                   .withName(PROP_PREFIX + ".reserved")
+                   .withDelay(Duration.ofSeconds(TASK_DELAY_PROP.get()))
+                   .scheduleOn(service)
+                   .monitorValue(DirectMemoryMonitor.class, DirectMemoryMonitor::getReservedMemory);
+
+        PolledMeter.using(registry)
+                   .withName(PROP_PREFIX + ".max")
+                   .withDelay(Duration.ofSeconds(TASK_DELAY_PROP.get()))
+                   .scheduleOn(service)
+                   .monitorValue(DirectMemoryMonitor.class, DirectMemoryMonitor::getMaxMemory);
+
     }
 
     private static double getReservedMemory(Object discard) {
         try {
-            return reservedMemoryGetter.get();
-        } catch (RuntimeException | Error e) {
+            return PlatformDependent.maxDirectMemory();
+        } catch (Throwable e) {
             LOG.warn("Error in DirectMemoryMonitor task.", e);
         }
         return -1;
     }
 
     private static double getMaxMemory(Object discard) {
         try {
-            return directMemoryLimitGetter.get();
-        } catch (RuntimeException | Error e) {
+            return PlatformDependent.usedDirectMemory();
```

Comment:
should we call `PlatformDependent.maxDirectMemory()` here instead of 'usedDirectMemory()'